text/0000-clarity-api-description.md

```diff
@@ -0,0 +1,77 @@
+- Start Date: 2015-06-18
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Summary
+
+Provide clarity and additional elements/properites in API Description namespace.
+
+# Motivation
+
+The purpose of this document is to make it easier for newcomers to understand the format and build tools around it.
+
+# Detailed design
+
+## Combine Namespaces
+
+We should combine the API Description namespace with the resource namespace.
+Right now they are separate and it would be clearer if they were not.
+
+## Elements in Attributes
+
+Refract allows for including elements within the attributes section of an element.
+Even though it’s possible, it may be helpful to avoid this in such an important namespace.
+Reason being, this will be one of the first places people interact with Refract, and to use complex features may be too much.
+
+## HREF Variables
+
+We should call them “hrefVariables” everywhere instead of parameters in places (specifically transitions).
+I think this will be more consistent and easier to understand for newcomers.
+This will make more sense in how we handle inheritance as well, because parameters inheriting from hrefVariables does not seem optimal.
+
+## Defining Classes
+
+Swagger provides a way to define tags that can be applied to resources throughout the document.
+Refract has the “class” attribute that we can use the same way, we only need a way to define a class name and provide the description for it.
+Of course, this won’t carry over to API Blueprint, but that’s OK.
```

Comment:
> Of course that assumes we use the tag title as its unique identifier.

I kind of struggle with this. On one hand, it makes sense because we would have to come with a unique ID for each tag. On the other hand, I get a funny feeling tying it to the human-readable title. I don't really feel strongly either way here.

I'm also a little iffy on having both category and tag in the spec. Seems like people have mental maps to what these are, and it would take some work to get your mind around the differences IMO. In other words, a resource can be in one and only one category, though it can be in many tags. (This is ignoring the fact the user could reference a resource with `ref`, so maybe I'm making a false assumption)

But I do really like this direction, and it seems like a necessity. We could leave this for a future change to the namespace so we can work this out more. @danielgtaylor @pksunkara